waltid-services/waltid-issuer-api/src/main/kotlin/id/walt/issuer/issuance/IssuerApi.kt

```diff
@@ -26,7 +27,12 @@ suspend fun createCredentialOfferUri(issuanceRequests: List<IssuanceRequest>): S
         OidcIssuance.issuanceRequestsToCredentialOfferBuilder(issuanceRequests)
 
     val issuanceSession = OidcApi.initializeCredentialOffer(
-        credentialOfferBuilder = credentialOfferBuilder, expiresIn = 5.minutes, allowPreAuthorized = true
+        credentialOfferBuilder = credentialOfferBuilder,
+        expiresIn = 5.minutes,
+        allowPreAuthorized = when (issuanceRequests[0].authenticationMethod) {
+            AuthenticationMethod.PRE_AUTHORIZED -> true
+            else -> false
+        }
     )
```

Comment:
This is a reasonable default expiration value, however, the client has no control. Perhaps we should consider adding an optional parameter with this as a default value?

Reply:
changed to method parameter with default value